contracts/UniversalRouter.sol

```diff
@@ -55,6 +76,198 @@ contract UniversalRouter is RouterImmutables, IUniversalRouter, Dispatcher, Rewa
         return command & Commands.FLAG_ALLOW_REVERT == 0;
     }
 
-    // To receive ETH from WETH and NFT protocols
+    /// @notice Decodes and executes the given command with the given inputs
```

Comment:
wait! I kind of miss having a separate dispatcher, felt so clean. why remove it?

Reply:
can it do `address(this).call(data)` ?

Reply:
yeah it can - thats what it already does if you look 😆 
but my point is that it breaks nice coding design if the parent contract is calling a function on the child contract. The parent contract should be designed in a way that does not rely on the design of the inheriting child contract.

Reply:
But you can still have the contracts be separated and use `address(this).call(data)`. Alternatively could you define an execute function inside the Dispatcher.sol contract that is virtual and in UniversalRouter just override it? 

I do think the separation of the contracts is pretty clean, but will leave it up to you guys to make the final call.

Reply:
> But you can still have the contracts be separated and use address(this).call(data). 

Yes sorry - I agree that this is _possible_, to me its just a bad programming style to have a contract call a function that it doesnt even know will ever exist

> Alternatively could you define an execute function inside the Dispatcher.sol contract that is virtual and in UniversalRouter just override it

This however is a much better solution because its then a function that the Dispatcher knows is guaranteed to exist.